app/renderer/ui/pages/softLogin/index.tsx

```diff
@@ -0,0 +1,140 @@
+import * as React from "react"
+import WalletSelection from "./steps/walletSelection"
+import WalletPasswordRequest from "./steps/walletPasswordRequest"
+import WalletUnlockInProgress from "./steps/walletUnlockInProgress"
+import { DispatchProps, StateProps } from "app/renderer/ui/containers/SoftLoginPage"
+
+const styles = require("./style.scss")
+
+// Steps in the soft login page
+enum Step {
+  walletSelection,
+  walletPasswordRequest,
+  walletUnlockInProgress,
+}
+
+/**
+ * SoftLoginPage component
+ *
+ * @export
+ * @class SoftLoginPage
+ * @extends {React.Component}
+ */
+export class SoftLogin extends React.Component<StateProps & DispatchProps> {
+  /**
+   * Method to get current step from state
+   * @returns {any}
+   */
+  private getStepFromState = (): Step => {
```

Comment:
Equivalent expression for the sake of [having a single exit point][exit]:
```ts
private guessStepFromState = (): Step =>
  // If unlock is in progress
  this.props.loginForm.unlockInProgress ?
    // Show walletUnlockInProgress
    Step.walletUnlockInProgress :
    // Else if there's a selected wallet ID
    this.props.loginForm.id ?
      // Show walletPasswordRequest
      Step.walletPasswordRequest :
      // Else show walletSelection
      Step.walletSelection
```
[exit]: https://www.tomdalling.com/blog/coding-tips/coding-tip-have-a-single-exit-point/

Reply:
great! thanks!